physics/noahmpdrv.F90

```diff
@@ -986,11 +1052,49 @@ subroutine noahmpdrv_run                                       &
             zvfun(i) = sqrt(tem1 * tem2)
             gdx=sqrt(garea(i))
 
+       if ( .not. do_mynnsfclay) then   !GFS sfcdiff
+
       call       stability                                                               &
         (zf(i), zvfun(i), gdx, virtual_temperature, vptemp,wind(i), z0_total, z0h_total, & 
          tvs1, con_g, thsfc_loc,                                                         &
          rb1(i), fm1(i), fh1(i), fm101(i), fh21(i), cm(i), ch(i), stress1(i), ustar1(i))
 
+       rmol1(i) = undefined  !not used in GFS sfcdif -> to satsify output
+       flhc1(i) = undefined
+       flqc1(i) = undefined
+
+        rah_total = max(1.0,1.0/( ch(i)*wind(i)) )
+        cah_total = density * con_cp /rah_total
+!       tskin(i) = sensible_heat_total/cah_total + temperature_forcing ! test to use combined ch and SH to backout Ts
+
+!        ch(i) = ch_vegetated * vegetation_frac + ch_bare_ground*(1.0-vegetation_frac)
```

Comment:
@barlage @RongqianYang-NOAA This is the one that causes the problem. ch_vegetated and ch_bare_ground are only available for non-glacial points. I have to comment this line out. Because we already agree ch should be used the one from third stability call not from the area averaged.